src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1308,6 +1308,7 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
                 catch (IOException ex)
```

Comment:
Discussed offline w/Safia. You should be able to add another catch block here for BadHttpRequestException and use the StatusCode it provides.

Reply:
Thought this might be where we land. I'll make the change.